helix-core/src/main/java/org/apache/helix/cloud/event/helix/HelixCloudEventListener.java

```diff
@@ -0,0 +1,150 @@
+      @Override
+      public CloudEventCallbackProperty.UserDefinedCallbackType toUserDefinedCallbackType() {
+        return CloudEventCallbackProperty.UserDefinedCallbackType.POST_ON_RESUME;
+      }
+    };
+
+    public CloudEventCallbackProperty.UserDefinedCallbackType toUserDefinedCallbackType() {
+      return null;
+    }
+  }
+
+  @Override
+  public void performAction(Object eventType, Object eventInfo) {
+    LOG.info("Received {} event, event info {}, timestamp {}. Acting on the event... "
+            + "Actor {}, based on callback implementation class {}.", ((EventType) eventType).name(),
+        eventInfo == null ? "N/A" : eventInfo.toString(), System.currentTimeMillis(), _helixManager,
+        _callbackImplClass.getClass().getCanonicalName());
+    Set<CloudEventCallbackProperty.HelixOperation> enabledHelixOperationSet =
+        _property.getEnabledHelixOperation();
+
+    if (eventType == EventType.ON_PAUSE) {
+      for (OnPauseOperations operation : OnPauseOperations.values()) {
+        switch (operation) {
+          case ENABLE_DISABLE_INSTANCE:
+            if (enabledHelixOperationSet
+                .contains(CloudEventCallbackProperty.HelixOperation.ENABLE_DISABLE_INSTANCE)) {
+              _callbackImplClass.disableInstance(_helixManager, eventInfo);
+            }
+            break;
+          case MAINTENANCE_MODE:
+            if (enabledHelixOperationSet
+                .contains(CloudEventCallbackProperty.HelixOperation.MAINTENANCE_MODE)) {
+              _callbackImplClass.enterMaintenanceMode(_helixManager, eventInfo);
+            }
+            break;
+          default:
+            if (operation.toUserDefinedCallbackType() != null) {
+              _property.getUserDefinedCallbackMap()
+                  .getOrDefault(operation.toUserDefinedCallbackType(), (manager, info) -> {
+                  }).accept(_helixManager, eventInfo);
+            }
+        }
+      }
+    } else if (eventType == EventType.ON_RESUME) {
+      for (OnResumeOperations operation : OnResumeOperations.values()) {
+        switch (operation) {
+          case ENABLE_DISABLE_INSTANCE:
+            if (enabledHelixOperationSet
+                .contains(CloudEventCallbackProperty.HelixOperation.ENABLE_DISABLE_INSTANCE)) {
+              _callbackImplClass.enableInstance(_helixManager, eventInfo);
+            }
+            break;
+          case MAINTENANCE_MODE:
+            if (enabledHelixOperationSet
+                .contains(CloudEventCallbackProperty.HelixOperation.MAINTENANCE_MODE)) {
+              _callbackImplClass.exitMaintenanceMode(_helixManager, eventInfo);
+            }
+            break;
+          default:
+            if (operation.toUserDefinedCallbackType() != null) {
+              _property.getUserDefinedCallbackMap()
+                  .getOrDefault(operation.toUserDefinedCallbackType(), (manager, info) -> {
+                  }).accept(_helixManager, eventInfo);
```

Comment:
for (OnPauseOperations operation : OpertionMap.values()) {
        switch (operation) {
          case ENABLE_DISABLE_INSTANCE:
            if (enabledHelixOperationSet
                .contains(CloudEventCallbackProperty.HelixOperation.ENABLE_DISABLE_INSTANCE)) {
              if (eventType == EventType.ON_PAUSE) {
                 _callbackImplClass.disableInstance(_helixManager, eventInfo);
              } else {
                 _callbackImplClass.enableInstance(_helixManager, eventInfo);
            }
            break;
          case MAINTENANCE_MODE:
            if (enabledHelixOperationSet
                .contains(CloudEventCallbackProperty.HelixOperation.MAINTENANCE_MODE)) {
              if (eventType == EventType.ON_PAUSE) {
              _callbackImplClass.enterMaintenanceMode(_helixManager, eventInfo);
              } else {
              _callbackImplClass.enterMaintenanceMode(_helixManager, eventInfo);
            }            }
            break;
          default:
            if (operation.toUserDefinedCallbackType() != null) {
              _property.getUserDefinedCallbackMap()
                  .getOrDefault(operation.toUserDefinedCallbackType(), (manager, info) -> {
                  }).accept(_helixManager, eventInfo);
            }
        }
      }

Reply:
We can reduce it to the specific line to do the check.

Reply:
Good point!! Updated.